src/plugins/discover/public/context_awareness/profile_providers/security/security_root_profile/profile.tsx

```diff
@@ -7,25 +7,80 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import React, { FunctionComponent, PropsWithChildren } from 'react';
+import { CustomCellRenderer, DataGridCellValueElementProps } from '@kbn/unified-data-table';
 import { RootProfileProvider, SolutionType } from '../../../profiles';
 import { ProfileProviderServices } from '../../profile_provider_services';
 import { SecurityProfileProviderFactory } from '../types';
+import { createCellRendererAccessor } from '../accessors/get_cell_renderer_accessor';
+import { createAppWrapperAccessor } from '../accessors/create_app_wrapper_accessor';
+
+interface SecurityRootProfileContext {
+  appWrapper?: FunctionComponent<PropsWithChildren<{}>>;
+  getCellRenderer?: (
+    fieldName: string
+  ) => FunctionComponent<DataGridCellValueElementProps> | undefined;
+}
+
+const EmptyAppWrapper: FunctionComponent<PropsWithChildren<{}>> = ({ children }) => <>{children}</>;
 
 export const createSecurityRootProfileProvider: SecurityProfileProviderFactory<
-  RootProfileProvider
-> = (services: ProfileProviderServices) => ({
-  profileId: 'security-root-profile',
-  isExperimental: true,
-  profile: {
-    getCellRenderers: (prev) => (params) => ({
-      ...prev(params),
-    }),
-  },
-  resolve: (params) => {
-    if (params.solutionNavId === SolutionType.Security) {
-      return { isMatch: true, context: { solutionType: SolutionType.Security } };
-    }
+  RootProfileProvider<SecurityRootProfileContext>
+> = (services: ProfileProviderServices) => {
+  const { discoverShared } = services;
+  const discoverFeaturesRegistry = discoverShared.features.registry;
+  const cellRendererFeature = discoverFeaturesRegistry.getById('security-solution-cell-renderer');
+  const appWrapperFeature = discoverFeaturesRegistry.getById('security-solution-app-wrapper');
+
+  return {
+    profileId: 'security-root-profile',
+    isExperimental: true,
+    profile: {
+      getRenderAppWrapper: (PrevWrapper, params) => {
+        const AppWrapper = params.context.appWrapper ?? EmptyAppWrapper;
+        return ({ children }) => (
+          <PrevWrapper>
+            <AppWrapper>{children}</AppWrapper>
+          </PrevWrapper>
+        );
+      },
+      getCellRenderers:
+        (prev, { context }) =>
+        (params) => {
+          const entries = prev(params);
+          const customCellRenderers = [
+            'host.name',
+            'user.name',
+            'source.ip',
+            'destination.ip',
+          ].reduce<CustomCellRenderer>((acc, fieldName) => {
+            acc[fieldName] = context.getCellRenderer?.(fieldName) ?? entries[fieldName];
```

Comment:
nit, but could you just `.forEach((fieldName) => { entries[fieldName] = context.getCellRenderer?.(fieldName) ?? entries[fieldName]; })`

Reply:
I like it. Fixed here:  7af29690d18